src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/ProjectView.xaml.cs

```diff
@@ -35,6 +47,207 @@ public ProjectView()
             _versions.ItemContainerStyle = style;
         }
 
+        private string _previousFilter;
+        private string PreviousFilter
+        {
+            get
+            {
+                if (_previousFilter != null) return _previousFilter;
+                return _versions.SelectedValue.ToString();
+            }
+            set
+            {
+                _previousFilter = value;
+                OnPropertyChanged(nameof(PreviousFilter));
+            }
+        }
+
+        private bool VersionsFilter(object o)
+        {
+            string ComboboxText = _versions.Text;
+            if (string.IsNullOrEmpty(_versions.Text)) return true;
+            if (ComboboxText.Length == 0 && ComboboxText.Equals("*", StringComparison.OrdinalIgnoreCase)) return true;
+            if ((ComboboxText.StartsWith("(", StringComparison.OrdinalIgnoreCase) || ComboboxText.StartsWith("[", StringComparison.OrdinalIgnoreCase)) &&
+               VersionRange.TryParse(ComboboxText, out VersionRange userRange))
+            {
+                if (o != null && NuGetVersion.TryParse(o.ToString(), out NuGetVersion userVersion))
+                {
+                    if (userRange.Satisfies(userVersion))
+                    {
+                        return true;
+                    }
+                    else { return false; }
+                }
+                else
+                {
+                    return false;
+                }
+            }
+            else
+            {
+                if (o != null && (o.ToString()).StartsWith(Regex.Replace(_versions.Text, @"[\*]", ""), StringComparison.OrdinalIgnoreCase)) return true;
```

Comment:
Similarly here, stripping away range characters feels like a helper method to me.